homeassistant/components/sensor/teleinfo.py

```diff
@@ -0,0 +1,148 @@
+"""
+Support for Teleinfo sensors.
+
+Teleinfo is a French specific protocol used in electricity smart meters.
+It provides real time information on power consumption, rates and current on
+a user accessible serial port.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.teleinfo/
+"""
+
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    ATTR_ATTRIBUTION, CONF_DEVICE, CONF_NAME, STATE_UNKNOWN)
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+
+REQUIREMENTS = ["kylin==0.5.0"]
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTRIBUTION = "Provided by EDF Teleinfo."
+
+DEFAULT_DEVICE = '/dev/ttyUSB0'
+DEFAULT_NAME = 'teleinfo'
+
+DATA_TELEINFO = "data_teleinfo"
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=1)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_DEVICE, default=DEFAULT_DEVICE): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+})
+
+TELEINFO_AVAILABLE_VALUES = ['HCHC', 'HCHP', 'IINST', 'IMAX', 'PAPP', 'ISOUSC']
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up for the Teleinfo device."""
+    from kylin import exceptions
+
+    teleinfo_data = None
+    try:
+        import kylin
```

Comment:
The import can probably be moved to the beginning of the function where we import the exceptions. Or is there a reason to put the import within the try... except?